### PR TITLE
[21.05] Add bref3 datatype

### PR DESCRIPTION
**Pull request by gallardoalba**
This PR includes the bref3 datatype, required for a new tool  https://github.com/galaxyproject/tools-iuc/pull/3760

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [x] Instructions for manual testing are as follows:

>>> from galaxy.datatypes.sniff import get_test_fname
>>> fname = get_test_fname('affy_v_agcc.cel')
>>> Bref3().sniff(fname)
False
>>> fname = get_test_fname('ref.bref3')
>>> Bref3().sniff(fname)
True

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

### CODE REVIEW

File: lib/galaxy/datatypes/binary.py [hunk 1]
```diff
@@ -236,6 +236,49 @@ def sniff(self, filename):
         return False
 
 
+class Bref3(Binary):
+    """Bref3 format is a binary format for storing phased, non-missing genotypes for a list of samples."""
```
Comment (bgruening) [inline lib/galaxy/datatypes/binary.py:240]: if you have a URL where this format is described please include it

**Review by bgruening — APPROVED** [review]

File: lib/galaxy/datatypes/binary.py (hunk 1 above)
Comment (gallardoalba) [reply lib/galaxy/datatypes/binary.py:240]: Yes

**Review by bgruening — APPROVED** [review]

File: lib/galaxy/datatypes/binary.py [hunk 2]
```diff
@@ -236,6 +236,49 @@ def sniff(self, filename):
...
+        >>> Bref3().sniff(fname)
+        True
+        """
+        try:
+            header = open(filename, 'rb').read(8)
+            if header == self._magic:
+                return True
+            return False
+        except Exception:
+            return False
```
Comment (mvdbeek) [inline lib/galaxy/datatypes/binary.py]:
```suggestion
    def sniff_prefix(self, sniff_prefix):
        return sniff_prefix.startswith_bytes(self._magic)
```

that should do the same without repeatedly opening the file for each dataset to sniff.

**Review by mvdbeek — APPROVED** [review]

**Comment by bgruening** [comment]
Thanks @gallardoalba @mvdbeek

**Pull request merged**